dotnet/src/SemanticKernel/Kernel.cs

```diff
@@ -301,28 +301,30 @@ public void Dispose()
     private static Dictionary<string, ISKFunction> ImportSkill(object skillInstance, string skillName, ILogger logger)
     {
         MethodInfo[] methods = skillInstance.GetType().GetMethods(BindingFlags.Static | BindingFlags.Instance | BindingFlags.Public);
-        logger.LogTrace("Importing skill name: {0}. Potential methods found: {1}", skillName, methods.Length);
-
+        logger.LogTrace($"Importing skill name: {skillName}. Potential methods found: {methods.Length}");
+    
         // Filter out non-SKFunctions and fail if two functions have the same name
         Dictionary<string, ISKFunction> result = new(StringComparer.OrdinalIgnoreCase);
         foreach (MethodInfo method in methods)
         {
             if (method.GetCustomAttribute<SKFunctionAttribute>() is not null)
             {
                 ISKFunction function = SKFunction.FromNativeMethod(method, skillInstance, skillName, logger);
-                if (result.ContainsKey(function.Name))
+                try
+                {
+                    result.Add(function.Name, function);
```

Comment:
What does this change fix? The PR's description doesn't describe any motivation for this.

The new code just seems less readable. What's the upside? Is there anything I am missing?